components/locale_core/src/parser/mod.rs

```diff
@@ -45,6 +45,14 @@ const fn get_current_subtag(slice: &[u8], idx: usize) -> (usize, usize) {
     (start, end)
 }
 
+pub const fn split_out_range(slice: &[u8], start: usize, end: usize) -> &[u8] {
+    assert!(start <= slice.len());
+    assert!(end <= slice.len());
+    assert!(start <= end);
+    // SAFETY: assertions and align = size = 1.
+    unsafe { core::slice::from_raw_parts(slice.as_ptr().add(start), end - start) }
+}
```

Comment:
`split_at_checked` is new in 1.80: https://github.com/rust-lang/rust/issues/85122

I think I prefer holding this until we can use that in MSRV so we don't introduce `unsafe` code.

Reply:
We did this because clients didn't have a safe way to do this at all. I'm assuming the majority of our clients will be on stable, so now the majority of our clients have a safe way to do this. One line more or less unsafe code to make this work isn't the end of the world. Also note that I have a follow-up PR that simplifies the unsafety, but it needs to be benched.

Reply:
ok, if this is just about getting rid of `_manual_slice` in public APIs, then just remove those functions without adding unsafe code?

Reply:
The parser calls the `_manual_slice` functions, because it needs to const-slice the input. I'm doing this in one location in the parser, and this is not the only unsafe code in this crate.

Reply:
Please add at least a comment here saying something like

```rust
// TODO(MSRV): Use slice_at_checked in Rust 1.80
```